lib/compiler-singlepass/src/machine_x64.rs

```diff
@@ -1626,17 +1626,13 @@ impl Machine for MachineX86_64 {
         self.used_gprs.insert(gpr);
     }
 
-    fn push_used_gpr(&mut self) -> usize {
-        let mut used_gprs = self.get_used_gprs();
-        used_gprs.sort();
+    fn push_used_gpr(&mut self, used_gprs: &Vec<GPR>) -> usize {
         for r in used_gprs.iter() {
```

Comment:
`HashSet` still has a randomized iteration order so this makes the codegen of singlepass non-deterministic (but still correct).

Reply:
True, but do we need to generate the same code on each run?

Reply:
Fair enough. I'll remove the hashmaps